src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -273,66 +273,57 @@ private static ChatRole ToChatRole(global::Azure.AI.Inference.ChatRole role) =>
         finishReason == CompletionsFinishReason.ToolCalls ? ChatFinishReason.ToolCalls :
         new(s);
 
+    private ChatCompletionsOptions CreateAzureAIOptions(IEnumerable<ChatMessage> chatContents, ChatOptions? options) =>
+        new(ToAzureAIInferenceChatMessages(chatContents))
+        {
+            Model = options?.ModelId ?? _metadata.DefaultModelId ?? throw new InvalidOperationException("No model id was provided when either constructing the client or in the chat options.")
+        };
+
     /// <summary>Converts an extensions options instance to an AzureAI options instance.</summary>
     private ChatCompletionsOptions ToAzureAIOptions(IEnumerable<ChatMessage> chatContents, ChatOptions? options)
     {
-        ChatCompletionsOptions result = new(ToAzureAIInferenceChatMessages(chatContents))
+        if (options is null)
         {
-            Model = options?.ModelId ?? _metadata.DefaultModelId ?? throw new InvalidOperationException("No model id was provided when either constructing the client or in the chat options.")
-        };
+            return CreateAzureAIOptions(chatContents, options);
+        }
 
-        if (options is not null)
+        if (options.RawRepresentation is ChatCompletionsOptions result)
         {
-            result.FrequencyPenalty = options.FrequencyPenalty;
-            result.MaxTokens = options.MaxOutputTokens;
-            result.NucleusSamplingFactor = options.TopP;
-            result.PresencePenalty = options.PresencePenalty;
-            result.Temperature = options.Temperature;
-            result.Seed = options.Seed;
-
-            if (options.StopSequences is { Count: > 0 } stopSequences)
-            {
-                foreach (string stopSequence in stopSequences)
-                {
-                    result.StopSequences.Add(stopSequence);
-                }
-            }
+            result.Messages = ToAzureAIInferenceChatMessages(chatContents).ToList();
```

Comment:
This seems surprising. `ChatCompletionsOptions` holds the list of messages? That makes it not seem like an "options" object at all but rather some kind of "request" representation. I know we don't own that type and didn't make that choice.

Mutating it like this would be surprising to me as an app developer. I would have expected I could reuse the same `chatOptions.RawRepresentation` instance across multiple requests, since in my expectation it represents some options that may be applicable to multiple requests. But if each request mutates it, we risk bugs.

Is there any way to clone a `ChatCompletionsOptions` before we mutate and use it?

Reply:
BTW I see this has been mentioned in the XML docs:

> Such implementations may mutate the supplied options instance further

However people won't necessarily understand that. It's pretty high-stakes since if a `ChatCompletionsOptions` was used across two different user sessions (which seems reasonable based on the naming), it could leak one user's messages into the other user's session.

Reply:
> but rather some kind of "request" representation

That's what it is: all of the arguments to CompleteAsync method, wrapped up into one object. It's not a great design.

> It's pretty high-stakes since if a ChatCompletionsOptions was used across two different user sessions

That's the design of ChatCompletionsOptions. Though I guess your point is that if folks don't realize the messages are included in "options", they might just set the options with their few knobs set and then reuse the same ChatOptions over and over.